fh-messaging/lib/fh_metrics_man.js

```diff
@@ -197,24 +197,67 @@ MetricsManager.prototype.deleteMetricsData = function(topic, date, callback) {
   // if multiple message managers pointed at the same config are running at the
   // same time
   var msgConfig = JSON.parse(JSON.stringify(self.mConfig));
-  var collectionName = topic + "_" + helpers.toYYYYMMDD(date);
-
   messageMan = new fhmsg.Messaging(msgConfig, self.mLogger);
   self.messageMans.push(messageMan);
+
   messageMan.database.on('tearUp', function() {
-    messageMan.database.db.dropCollection(collectionName, callback);
-    //remove all the data that is older than the x number of days from the intermediate tables
-    async.each(['transactionsdesttemp', 'transactionsgeotemp'], function(tempCol, cb) {
-      var daysToCalculate = self.mConfig.agenda.jobs.metrics_rollup_job.daysToKeep || 31;
-      var beforeDate = new Date(date.getFullYear(), date.getMonth(), date.getDate() - daysToCalculate);
-      var query = {"_id.ts":{$lt: beforeDate.getTime()}};
-      messageMan.database.db.collection(tempCol, function(err, collection) {
+    // Pruning for fhact
+    var daysToCalculate = self.mConfig.agenda.jobs.metrics_rollup_job.daysToKeep || 31;
+    var pruning_date = new Date().getTime() - ( daysToCalculate * 24 * 60 * 60 * 1000 );
+    var prune_collection_names = [];
+
+    // Retrieve the list of collections from the database
+    messageMan.database.db.listCollections({}).toArray( function(err, items) {
```

Comment:
Why you're getting the full collection from the database instead of using the query as it was before? Note that now you take the full data there and after you are filtering the data by ` if ( collection_date.getTime() < parseInt(pruning_date) ) {`. I don't think it is a good idea. IMHO you should instead of doing an if with the condition use it as a query conditional. Also, you are testing it locally which shows doesn't have data a lot and may you face the opposite in production which can cause many issues.

Reply:
Hmm the problem is that in production we have collections randomly left over when the pruning jobs fail, so this is to prune all of those collections older than the:

```
var daysToCalculate = self.mConfig.agenda.jobs.metrics_rollup_job.daysToKeep || 31;
```

Reply:
@davidkirwan. Oks, but is not possible here add in the query the conditional? (collection_date.getTime() < parseInt(pruning_date) )

Reply:
The collections don't have a reliable date field other than in the name of the collection from what I can see.

Reply:
Usuing a regex I can retrieve a list of collections which match the particular topic being pruned:
```
messageMan.database.db.listCollections({name: RegExp("^" + topic + "_")}).toArray( function(err, items)
```

@camilamacedo86 can you take another look at the latest PR now cheers.